### PR TITLE
feat(shorebird_cli): add patch ios-framework-alpha command

### PR DESCRIPTION
**Pull request by bryanoltman**
## Description

Adds the `patch ios-framework-alpha` to patch iOS frameworks in an add-to-app scenario.

Part of https://github.com/shorebirdtech/shorebird/issues/896

## Type of Change

- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

File: packages/shorebird_cli/lib/src/commands/patch/patch_ios_framework_command.dart [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+  @override
+  String get name => 'ios-framework-alpha';
+
+  @override
+  String get description =>
+      'Publish new patches for a specific iOS framework release to Shorebird.';
+
+  @override
+  Future<int> run() async {
+    if (!platform.isMacOS) {
```
Comment (felangel) [inline packages/shorebird_cli/lib/src/commands/patch/patch_ios_framework_command.dart]: minor but might be cleaner to add this to `validatePreconditions` like:

```dart
await validatePreconditions(
  supportedPlatforms: [Platform.macOS],
  ...
);
```

File: packages/shorebird_cli/test/src/commands/patch/patch_ios_framework_command_test.dart [hunk 2]
```diff
@@ -0,0 +1,572 @@
...
+          ),
+        ),
+      ).called(1);
+      verify(() => logger.success('\n✅ Published Patch!')).called(1);
+      expect(exitCode, ExitCode.success.code);
+    });
+
+    test(
+        'succeeds when patch is successful '
+        'with flavors and target', () async {
```
Comment (felangel) [inline packages/shorebird_cli/test/src/commands/patch/patch_ios_framework_command_test.dart]: can we either support flavor/target as options or remove this?

Comment (bryanoltman) [reply packages/shorebird_cli/test/src/commands/patch/patch_ios_framework_command_test.dart]: Done.

File: packages/shorebird_cli/lib/src/commands/patch/patch_ios_framework_command.dart (hunk 1 above)
Comment (bryanoltman) [reply packages/shorebird_cli/lib/src/commands/patch/patch_ios_framework_command.dart]: Done.

**Review by felangel — APPROVED** [review]

File: packages/shorebird_cli/lib/src/commands/patch/patch_ios_framework_command.dart [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+    with
+        ShorebirdConfigMixin,
+        ShorebirdValidationMixin,
+        ShorebirdBuildMixin,
+        ShorebirdArtifactMixin {
+  PatchIosFrameworkCommand({
+    HashFunction? hashFn,
+  }) : _hashFn = hashFn ?? ((m) => sha256.convert(m).toString()) {
+    argParser
+      ..addOption(
```
Comment (felangel) [inline packages/shorebird_cli/lib/src/commands/patch/patch_ios_framework_command.dart:30]: @bryanoltman can we add flavor support now before merging this?

Comment (bryanoltman) [reply packages/shorebird_cli/lib/src/commands/patch/patch_ios_framework_command.dart:30]: Ha, sure! I thought I merged this 😳

Comment (bryanoltman) [reply packages/shorebird_cli/lib/src/commands/patch/patch_ios_framework_command.dart:30]: Done

**Review by felangel — APPROVED** [review]

**Pull request merged**